### ISSUE TITLE
Personal scores & scoring criteria

### ISSUE THREAD

**Issue by strikef** [snu-sf-class/swpp202201#97]
## Sorry for the delayed grade announcement
We were initially planning on announcing everyone's grade on 28th, but we acknowledged an issue just before the announcement and had to revisit some of your scores.

Fortunately we just finished the revision, but it may take another day or two for your grades to be announced. Therefore we decided to upload everyone's score ASAP, so that we can start taking your claims as soon as possible.

### Assignments (Max. 25)  
In this course each of you had to submit 4 assignments. They account for 25% of your course score.
The first assignment was relatively easy, while the last assignment was quite difficult and even involved competition. We decided to give more weight to more difficult assignments, as the table below.

|  | Assignment 1 | Assignment 2 | Assignment 3 | Assignment 4 | Sum |
| --- | --- | --- | --- | --- | --- |
| Assignment score weight | 15 | 20 | 25 | 40 | 100 |
| Course score weight  | 3.75 | 5 | 6.25 | 10 | 25 |

### Team project (Max. 40)
We graded every PRs and reviews you wrote during each sprint. Each PR and review score has even weight regardless of the sprints, and they account for 40% of your course score. **However, we did not grade the PRs and reviews if they fall into any one of the following categories:**
* **PRs without sprint tags** (e.g. those that does not have [Sprint N] in the PR title)
* NFC and backend update PRs (but we graded them if they are mixed with individual's work)
* PRs during wrap-up period

Failing to satisfy the sprint requirements (more than one NFCs, mixed topics, **unsquashed PRs**, etc) were subject to deduction, while those managed to satisfy all requirements received a small bonus per each sprint.

|  | Sprint 1 PR | Sprint 1 Review | Sprint 2 PR | Sprint 2 Review | Sprint 3 PR | Sprint 3 Review | Sum |
| --- | --- | --- | --- | --- | --- | --- | --- |
| Project score weight | 16.66 | 16.66 | 16.66 | 16.66 | 16.66 | 16.66 | 100 |
| Course score weight  | 6.66 | 6.66 | 6.66 | 6.66 | 6.66 | 6.66 | 40 |

### Sprint documents (Max. 10)
Each team had to submit total of 7 sprint documents during the project period. They account for 10% of your course score.
We gave you full credit if your documents did contain everything we depicted in the project document.
| | Planning | Sprint 1 req/spec | Sprint 1 progress | Sprint 2 req/spec | Sprint 2 progress | Sprint 3 req/spec | Sprint 3 progress | Sum |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Documents score weight | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 100 |
| Course score weight  | 1.43 | 1.43 | 1.43 | 1.43 | 1.43 | 1.43 | 1.43 | 10 |

### Competition (Max. 25)
We already notified each team's competition score on [this issue](https://github.com/snu-sf-class/swpp202201/issues/96). But we normalized these scores as following:
> St: your team compiler's score, Sb: base compiler's score (206.0), Sh: the best performing compiler's score
`MAX(St, Sb) / Sh * 25`

Therefore if your compiler showed the best performance, you'll get 25 points. However, even if your compiler did not perform any better than the base compiler, you still get about 15 points.
**However, we gave 0 points to those who showed absolutely no contribution to the project**. This is very heavy penalty, so we gave it only when we were absolutely certain that they did not contribute to the project at all.

## TL; DR
Okay, here's everyone's score.
[grading-table.pdf](https://github.com/snu-sf-class/swpp202201/files/9018828/grading-table.pdf)

## Claims
We'll take claims for your scores (and grades, once it is announced) via TA e-mail until July 3rd (Sun), KST. We may not be able to reply during the weekends, but we will reply as long as your mail has arrived before the deadline.
**Please write `[SWPP claim]` in front of your subject (제목), or we will not take that claim**. You may send your claim either in English or Korean.
